base/setup/lib/utils/filesup.c

```diff
@@ -27,35 +27,30 @@
 static
 NTSTATUS
 SetupCreateSingleDirectory(
-    IN PCWSTR DirectoryName)
+    _In_ PCUNICODE_STRING DirectoryName)
 {
+    NTSTATUS Status;
+    UNICODE_STRING PathName = *DirectoryName;
     OBJECT_ATTRIBUTES ObjectAttributes;
     IO_STATUS_BLOCK IoStatusBlock;
-    UNICODE_STRING PathName;
     HANDLE DirectoryHandle;
-    NTSTATUS Status;
-
-    if (!RtlCreateUnicodeString(&PathName, DirectoryName))
-        return STATUS_NO_MEMORY;
 
-    if (PathName.Length > sizeof(WCHAR) &&
-        PathName.Buffer[PathName.Length / sizeof(WCHAR) - 2] == L'\\' &&
+    if (PathName.Length >= 2 * sizeof(WCHAR) &&
```

Comment:
Sadly there is no 'postfix' alternative to `RtlPrefixUnicodeString`, otherwise that could be used.

I did roll my own, but I don't think that would make sense to add here:
https://github.com/reactos/reactos/blob/master/dll/appcompat/shims/layer/vmhorizon.c#L85

Reply:
The existing code here looks simple enough (since we are just here dealing with checking one or two characters) so that we can avoid the usage of any sort of function here.